### PR TITLE
Create new text input component and replace all existing ones with it

### PR DESCRIPTION
**Pull request by axbarsan**
Towards https://github.com/giantswarm/giantswarm/issues/11004 and towards the `grommet` migration

This PR adds a new `TextInput` component, useful for creating text fields. It also includes different Storybook stories for it, and snapshot tests (so we could know if anything changes in the `grommet` components, which might break ours).

Also, in this PR, all the `text` inputs in the app have been converted to the new component. Some of them look slightly different now. Also, in some cases, the labels contained `:` at the end, where in others they didn't. Now none of them have `:` at the end.

Everything has been thoroughly tested, but you can still check for yourself 😜 

### Preview

<details>
<summary>With label</summary>

image

</details>

<details>
<summary>With error</summary>

image

</details>

<details>
<summary>With info</summary>

image

</details>

<details>
<summary>With help</summary>

image

</details>

<details>
<summary>Disabled</summary>

image

</details>

### CODE REVIEW

File: src/components/AccountSettings/__tests__/ChangePasswordForm.js [hunk 1]
```diff
@@ -190,9 +190,9 @@ describe('ChangePasswordForm', () => {
 
     triggerInputChange(confirmNewPasswordInput, newPassword);
 
-    // Check if set button is present and disabled
```
Comment (oponder) [inline src/components/AccountSettings/__tests__/ChangePasswordForm.js:193]: Interesting! so this test was actually never really working right?

**Review by oponder — COMMENTED** [review]
LGTM, epic!
We did lose the background colour of the inputs everywhere, which I kinda liked, but am not strongly attached to. Was that on purpose?

Before: 

Screenshot 2021-02-09 at 8 48 03 PM

After:

Screenshot 2021-02-09 at 8 48 02 PM

File: src/components/AccountSettings/__tests__/ChangePasswordForm.js (hunk 1 above)
Comment (axbarsan) [reply src/components/AccountSettings/__tests__/ChangePasswordForm.js:193]: Exactly 😅

**Comment by axbarsan** [comment]
> We did lose the background colour of the inputs everywhere, which I kinda liked, but am not strongly attached to. Was that on purpose?

Yeah, that was on purpose, since the color used before was not in the theme, and I wanted to make use of less colors this time around (we currently have over 40 different colors in our theme).

However, if you think that inputs look better with the background color, I can add it back in 👍🏻

**Comment by oponder** [comment]
Im fine if we consolidate some of the colors in the theme, maybe find one currently in the theme that is close?

**Comment by axbarsan** [comment]
> Im fine if we consolidate some of the colors in the theme, maybe find one currently in the theme that is close?

I've updated the screenshots, please let me know what you think.

**Review by oponder — APPROVED** [review]
thx man!

**Pull request merged**